src/hooks/useStorage.ts

```diff
@@ -0,0 +1,34 @@
+import { useCallback, useState } from 'react';
+
+export const useStorage = <T>(
```

Comment:
we should only create hooks if we interact with the react ecosystem. I think this could have been a simple function

Reply:
Please look into it. It behaves as a `useState`, but keeps the data in storage. It reacts to any changes of the stored value